include/utils/error.hpp

```diff
@@ -24,13 +24,15 @@ RequestClearErrorOption string_to_request_clear_error_option(const std::string&
 
 class NotValidErrorType : public std::exception {
 public:
-    explicit NotValidErrorType(const std::string& error_type_) : error_type(error_type_){};
-    virtual const char* what() const throw() {
-        return fmt::format("Error type '{}' is not valid.", error_type).c_str();
+    explicit NotValidErrorType(const std::string& error_type_) :
+        error_type(error_type_), explanatory_string(fmt::format("Error type '{}' is not valid.", error_type)){};
+    const char* what() const noexcept override {
+        return explanatory_string.c_str();
     }
 
 private:
     const std::string error_type;
```

Comment:
```suggestion
    const std::string error_type;

private:
```

Reply:
I'm not so sure about making this member public, should probably be a getter. Or we remove it for now if there's no usage

Reply:
I would remove it here, if it will be needed, it can just be added